### PR TITLE
adding_dagmc_v3.2.3_back

### PR DESCRIPTION
**Pull request by shimwell**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR attempts to add DAGMC back into the openmc conda build. DAGMC was removed during the 0.14.0 release of open as we were seeing some build issues. However now that DAGMC version 3.2.3 has been released we should have another go at getting DAGMC back.

DAGMC 3.2.3 was recently released with [this ](https://github.com/conda-forge/dagmc-feedstock/pull/26)PR

### CODE REVIEW

**Comment by shimwell** [comment]
@conda-forge-admin, please rerender

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+dagmc:
+  - nodagmc
+  - dagmc
+
```
Comment (shimwell) [inline recipe/conda_build_config.yaml:4]:
```suggestion
```

**Comment by shimwell** [comment]
perhaps have to wait for the new dagmc to be found, ci error message is

```
- Encountered problems while solving:
-   - package dagmc-3.2.3-mpi_mpich_py310hc89fbf8_0 requires hdf5 >=1.14.2,<1.14.3.0a0 mpi_mpich_*, but none of the providers can be installed
- 
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -4,6 +4,14 @@
 {% set sha256 = "20ac92cc8d37a8c1180ae2df9f1dd1158f188cbf724ae64c9642db6199f3ee3f" %}
 {% set build = 1 %}
 
+# ensure dagmc is defined (needed for conda-smithy recipe-lint)
+{% set dagmc = dagmc or 'nodagmc' %}
+
```
Comment (shimwell) [inline recipe/meta.yaml:9]:
```suggestion
```

**Comment by shimwell** [comment]
Looks like a version of hdf5 that satisfies all the dependencies can't be found

**Comment by paulromano** [comment]
Looks like the failure has something to do with the fact that 1.14.2 and 1.14.3 are supposed to be API compatible, but the version range dependencies were not updated on all packages correctly?
https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/611#issuecomment-1847712560

Easy fix should just be to update the HDF5 pinning on dagmc-feedstock, which I've just submitted a PR for

File: recipe/meta.yaml [hunk 3]
```diff
@@ -4,6 +4,13 @@
 {% set sha256 = "20ac92cc8d37a8c1180ae2df9f1dd1158f188cbf724ae64c9642db6199f3ee3f" %}
 {% set build = 1 %}
 
+# ensure dagmc is defined (needed for conda-smithy recipe-lint)
+{% set dagmc = dagmc or 'nodagmc' %}
+{% if dagmc == 'dagmc' %}
```
Comment (shimwell) [inline recipe/meta.yaml:9]:
```suggestion
{% if dagmc == 'dagmc' %}

```

**Comment by shimwell** [comment]
Super super, that appears to have worked @paulromano 

Is this OK to merge?

**Comment by paulromano** [comment]
All good; thanks @shimwell!

**Pull request merged**